### PR TITLE
In some lfs server implementations, they require the ref attribute.

### LINKED ISSUE #32611: mirror: LFS client version too old
### Description

`epository/migrate.go:171:MigrateRepositoryGitData() [E] Failed to store missing LFS objects for repository: LFS client version too old, please upgrade your LFS client greater than v2.4.0`

I install the latest git-lfs version in the container, but it did not work.
image

#30913 

### Gitea Version

1.22.3

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Screenshots

image

### Git Version

2.45.2

### Operating System

Docker

### How are you running Gitea?

docker compose

### Database

MySQL/MariaDB

### PR DESCRIPTION
**Pull request by hiifong**
Fix: #32611

In some lfs server implementations, they require the ref attribute. Note: Added in v2.4.

https://github.com/git-lfs/git-lfs/blob/a32a02b44bf8a511aa14f047627c49e1a7fd5021/docs/api/batch.md?plain=1#L37

image

### CODE REVIEW

**Comment by wxiaoguang** [comment]
What if an object is not stored in "default branch"?

* Before: objects in all branches should be downloaded
* After: only objects in default branch could be downloaded?

**Comment by hiifong** [comment]
> What if an object is not stored in "default branch"?
> 
> * Before: objects in all branches should be downloaded
> * After: only objects in default branch could be downloaded?

Updated

**Review by delvh — APPROVED** [review]

File: modules/lfs/http_client.go [hunk 1]
```diff
@@ -72,7 +72,7 @@ func (c *HTTPClient) batch(ctx context.Context, operation string, objects []Poin
 
 	url := fmt.Sprintf("%s/objects/batch", c.endpoint)
 
-	request := &BatchRequest{operation, c.transferNames(), nil, objects}
+	request := &BatchRequest{operation, c.transferNames(), &Reference{}, objects}
```
Comment (wxiaoguang) [inline modules/lfs/http_client.go:78]:
```suggestion
	// `ref` is an "optional object describing the server ref that the objects belong to"
	// but some (incorrect) lfs servers require it, so maybe adding an empty ref here doesn't break the correct ones.
	// https://github.com/git-lfs/git-lfs/blob/a32a02b44bf8a511aa14f047627c49e1a7fd5021/docs/api/batch.md?plain=1#L37
	request := &BatchRequest{operation, c.transferNames(), &Reference{}, objects}
```

**Review by wxiaoguang — APPROVED** [review]

**Comment by wxiaoguang** [comment]
We need to revert this, it causes more problems like "Unable to migrrate repos from Hugging Face #33453"

**Comment by lunny** [comment]
#32611 need to be reopen

**Pull request merged**